src/meta_schedule/database/memory_database.cc

```diff
@@ -71,7 +71,7 @@ class MemoryDatabaseNode : public DatabaseNode {
       if (run_secs.empty()) {
         continue;
       }
-      if (record->workload.same_as(workload)) {
+      if (WorkloadEqual(GetModuleEquality())(record->workload, workload)) {
```

Comment:
What does it fix? They don't seem equivalent (pointer vs structural equality)

Reply:
Here we wanted to find the tuning records for the structrually equal module, but here instead of doing WorkloadEquality comparsion, it's directly doing pointer comparison, which doesn't seem to be correct.